### PR TITLE
Add 'optional' command line options

### PR DESCRIPTION
**Pull request by ska80**
Also combine `--ide-mode-socket` and `--ide-mode-socket-with` options into one single `--ide-mode-socket` option.

### CODE REVIEW

File: src/Idris/CommandLine.idr [hunk 1]
```diff
@@ -146,27 +149,27 @@ options = [MkOpt ["--check", "-c"] [] [CheckOnly]
...
 
            MkOpt [] [] [] Nothing,
            MkOpt ["--ide-mode"] [] [IdeMode]
               (Just "Run the REPL with machine-readable syntax"),
-           MkOpt ["--ide-mode-socket"] [] [IdeModeSocket $ formatSocketAddress (ideSocketModeAddress [])]
-              (Just $ "Run the ide socket mode on default host and port (" ++ formatSocketAddress (ideSocketModeAddress []) ++ ")"),
-           MkOpt ["--ide-mode-socket-with"] ["host:port"] (\hp => [IdeModeSocket hp])
-              (Just "Run the ide socket mode on given host and port"),
+           MkOpt ["--ide-mode-socket"] [Optional "host:port"]
+                 (\hp => [IdeModeSocket $ maybe (formatSocketAddress (ideSocketModeAddress [])) id hp])
```
Comment (gallais) [inline src/Idris/CommandLine.idr:167]:
```suggestion
                 (\hp => [IdeModeSocket $ fromMaybe (formatSocketAddress (ideSocketModeAddress [])) hp])
```

File: src/Idris/CommandLine.idr [hunk 2]
```diff
@@ -260,7 +260,7 @@ processArgs flag (Optional a :: as) [] f =
 processArgs flag (Required a :: as) (x :: xs) f =
   processArgs flag as xs (f x)
 processArgs flag (Optional a :: as) (x :: xs) f =
-  processArgs flag as xs (f (Just x))
+  processArgs flag as xs (f $ toMaybe (not (any (\p => isPrefixOf p x) ["-", "--"])) x)
```
Comment (gallais) [inline src/Idris/CommandLine.idr:263]: If `--` is a prefix then so is `-`. So it's enough to check for `-` only.

Comment (ska80) [reply src/Idris/CommandLine.idr:263]: Makes sense.  Thanks, I will fix it.

**Comment by gallais** [comment]
I see multiple merges. Do you think it's ready @ska80?
I'm basically just waiting for you to say so, after seeing that
you were still adding some nice refactoring & a CHANGELOG entry.

**Comment by ska80** [comment]
> I see multiple merges. Do you think it's ready @ska80?
> I'm basically just waiting for you to say so, after seeing that
> you were still adding some nice refactoring & a CHANGELOG entry.

Yes, it is ready.  Sorry for not telling you.

**Pull request merged**